src/compiler/tsbuild.ts

```diff
@@ -0,0 +1,1222 @@
+            return childToParents[normalizePath(childConfigFileName)] || [];
+        }
+
+        function getKeys(): ReadonlyArray<ResolvedConfigFileName> {
+            return allKeys;
+        }
+
+        function addEntry(mapToAddTo: typeof childToParents | typeof parentToChildren, key: ResolvedConfigFileName, element: ResolvedConfigFileName) {
+            key = normalizePath(key) as ResolvedConfigFileName;
+            element = normalizePath(element) as ResolvedConfigFileName;
+            const arr = (mapToAddTo[key] = mapToAddTo[key] || []);
+            if (arr.indexOf(element) < 0) {
+                arr.push(element);
+            }
+            if (allKeys.indexOf(key) < 0) allKeys.push(key);
+            if (allKeys.indexOf(element) < 0) allKeys.push(element);
+        }
+
+        return {
+            addReference,
+            getReferencesTo,
+            getReferencesOf,
+            getKeys
+        };
+    }
+
+    function getOutputDeclarationFileName(inputFileName: string, configFile: ParsedCommandLine) {
+        const relativePath = getRelativePathFromDirectory(rootDirOfOptions(configFile.options, configFile.options.configFilePath!), inputFileName, /*ignoreCase*/ true);
+        const outputPath = resolvePath(configFile.options.declarationDir || configFile.options.outDir || getDirectoryPath(configFile.options.configFilePath!), relativePath);
+        return changeExtension(outputPath, ".d.ts");
+    }
+
+    function getOutputJavaScriptFileName(inputFileName: string, configFile: ParsedCommandLine) {
+        const relativePath = getRelativePathFromDirectory(rootDirOfOptions(configFile.options, configFile.options.configFilePath!), inputFileName, /*ignoreCase*/ true);
+        const outputPath = resolvePath(configFile.options.outDir || getDirectoryPath(configFile.options.configFilePath!), relativePath);
+        return changeExtension(outputPath, (fileExtensionIs(inputFileName, ".tsx") && configFile.options.jsx === JsxEmit.Preserve) ? ".jsx" : ".js");
+    }
+
+    function getOutputFileNames(inputFileName: string, configFile: ParsedCommandLine): ReadonlyArray<string> {
+        if (configFile.options.outFile) {
+            return emptyArray;
+        }
+
+        const outputs: string[] = [];
+        outputs.push(getOutputJavaScriptFileName(inputFileName, configFile));
+        if (configFile.options.declaration) {
+            const dts = outputs.push(getOutputDeclarationFileName(inputFileName, configFile));
+            if (configFile.options.declarationMap) {
+                outputs.push(dts + ".map");
+            }
+        }
+        return outputs;
+    }
+
+    function getOutFileOutputs(project: ParsedCommandLine): ReadonlyArray<string> {
+        if (!project.options.outFile) {
+            throw new Error("Assert - outFile must be set");
+        }
+        const outputs: string[] = [];
+        outputs.push(project.options.outFile);

```

Comment:
this looks a lot like getOutputPathsFor. just use that instead.

Reply:
Same problem here; those functions were written taking much wider inputs than required or present